src/cdk-experimental/table/virtual-rows.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  Directive,
+  ElementRef,
+  Inject,
+  Input,
+  OnDestroy,
+  SkipSelf,
+} from '@angular/core';
+import {
+  _RecycleViewRepeaterStrategy,
+  _VIEW_REPEATER_STRATEGY,
+  isDataSource,
+  ListRange
+} from '@angular/cdk/collections';
+import {
+  CdkTable,
+  DataSource,
+  RenderRow,
+  RowContext,
+  STICKY_POSITIONING_LISTENER,
+  StickyPositioningListener,
+  StickyUpdate
+} from '@angular/cdk/table';
+import {
+  combineLatest,
+  isObservable,
+  Observable,
+  of as observableOf, OperatorFunction,
+  ReplaySubject,
+  Subject,
+} from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  shareReplay,
+  skip,
+  switchMap,
+  takeUntil
+} from 'rxjs/operators';
+import {
+  CdkVirtualScrollRepeater,
+  CdkVirtualScrollViewport,
+  VIRTUAL_SCROLL_STRATEGY,
+  VirtualScrollStrategy
+} from '@angular/cdk/scrolling';
+
+
+/**
+ * Union of the types that can be set as the data source for a `CdkTable`.
+ * @docs-private
+ */
+type CdkTableDataSourceInput<T> =
```

Comment:
Couldn't you import this from `cdk/table`?

Reply:
Done. I also renamed `CdkTableDataSourceInput` to `_CdkTableDataSourceInput`.